tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -543,6 +545,16 @@ spv::LoopControlMask SPIRVEmitter::translateLoopAttribute(const Attr &attr) {
   return spv::LoopControlMask::MaskNone;
 }
 
+void SPIRVEmitter::doDiscardStmt(const DiscardStmt *discardStmt) {
+  assert(!theBuilder.isCurrentBasicBlockTerminated());
+  theBuilder.createKill();
+  if (!isLastStmtBeforeControlFlowBranching(astContext, discardStmt)) {
```

Comment:
I see this code snippet appears also after `continue` and `break`. Maybe we can create a method for it to avoid duplication?